compiler/src/dotty/tools/dotc/core/classfile/ClassfileConstants.scala

```diff
@@ -375,6 +375,6 @@ object ClassfileConstants {
     override def baseFlags(jflags: Int) = if ((jflags & JAVA_ACC_FINAL) == 0) Mutable else EmptyFlags
   }
   val methodTranslation: FlagTranslation = new FlagTranslation {
-    override def baseFlags(jflags: Int) = if ((jflags & JAVA_ACC_BRIDGE) != 0) Bridge else EmptyFlags
+    override def baseFlags(jflags: Int) = if ((jflags & JAVA_ACC_BRIDGE) != 0) BridgeArtifact else EmptyFlags
```

Comment:
This shouldn't be needed since all bridges are emitted with the ACC_SYNTHETIC flag which corresponds to Artifact in Scala. The problem seems to be that we don't actually set the Artifact flag for them: https://github.com/lampepfl/dotty/blob/9dbade3e3c08a81a2b2744d3afa5a8434bd0cdba/compiler/src/dotty/tools/dotc/core/classfile/ClassfileConstants.scala#L344 compare with https://github.com/scala/scala/blob/a26aeb7472ae2d2bab331bfcf9f785c3c7d1c18a/src/reflect/scala/reflect/internal/ClassfileConstants.scala#L350

Reply:
Why do they set it explicitly then at https://github.com/scala/scala/blob/a26aeb7472ae2d2bab331bfcf9f785c3c7d1c18a/src/reflect/scala/reflect/internal/ClassfileConstants.scala#L382?

Reply:
No particular reason I think, since no one ever generates non-synthetic bridges it doesn't make a difference (but non-bridge synthetic do exist so fixing that is important)

Reply:
Ok, seems to be working without that line